src/frp/provideDuration.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+import Observable from './Observable';
+
+const provideDuration = s$ => to$ => {
+    return (...args) => {
+        return Observable.create(sub => {
+            const start = Date.now();
+            to$(...args).subscribe(x => {
+                s$.next(Date.now() - start);
+                sub.next(x);
+            });
+        });
+    };
+};
```

Comment:
Very minor suggestion: this may be a candidate for using `multicast` with a selector function. Ex:

```es6
const provideDuration = s$ => to$ => {
    return (...args) => {
        const to$Inst = to$(...args);
        return to$Inst.multicast(
            // Create a Subject instance which multicast
            // will use to subscribe to to$Inst when
            // someone subscribes to multicast
            () =>         new Subject(),
            (to$Shared /* <-- is ^ this guy */) => {

                const start = Date.now();

                const duration$ = to$Shared
                    .map(() => Date.now() - start)
                    .do(s$) // <-- emit durations to the supplied Subject.
                    //             todo: should $s hear about error/complete from to$Inst?
                    //             If so, this is fine. If not, `do(s$.next.bind(s$))` is the move.
                    .ignoreElements(); // <-- don't send duration values to outer subscriber

                return duration$ // <-- ensure duration$ is subscribed to
                    .merge(to$Shared) // <-- pipe all to$Inst values to the outer subscriber
            }
        );
    }
};
```